### PR TITLE
feat(CapMan): Pass `tenant_ids` to Snuba

### PR DESCRIPTION
**Pull request by rahul-kumar-saini**
### Overview
- As part of the [Capacity Management](https://www.notion.so/sentry/Clickhouse-Capacity-Management-df585f5b7ae4453a9e635e1520b4a211) project over on [S<sup>3</sup>](https://www.notion.so/sentry/Search-Storage-Streaming-a5d224329fe148e4b27ff228339258b1), we're working on getting more information out of per Snuba Request in order to properly categorize and eventually set precise limits on API usage by a number of different variables (we're calling them tenants)

### Changes
- The first two tenant IDs we're implementing are `organization_id` and `referrer` (referrer already existed but this PR adds it to the dictionary)
- This PR updates the Snuba SDK version in Sentry and starts the process of consolidating these tenants into a single dictionary called `tenant_ids` per request
- I've extracted `organization_id` from some easy use cases of the Snuba API and will be working on more complicated ones in future PRs

### Notes
- Eventually, we will need Sentry's backend teams to help figure out how to get more of these tenant IDs into the requests if/when things get too complicated

### CODE REVIEW

**Review by volokluev — APPROVED** [review]
just one question

File: src/sentry/tsdb/snuba.py [hunk 1]
```diff
@@ -448,10 +456,9 @@ def __get_data_snql(
                     granularity=Granularity(rollup),
                     limit=Limit(limit),
                 ),
+                tenant_ids=tenant_ids or dict(),
```
Comment (evanh) [inline src/sentry/tsdb/snuba.py:459]: Why do you pass an empty `dict` here? You have the referrer at least.

**Review by JoshFerge — APPROVED** [review]
seems safe. couple questions on consistency:

- why are some referrers defined in a constants file and others not?
- why do some referrers use underscores and others dashes?

**Comment by rahul-kumar-saini** [comment]
@JoshFerge 
> seems safe. couple questions on consistency:
> 
> * why are some referrers defined in a constants file and others not?
> * why do some referrers use underscores and others dashes?

I only refactored existing referrers within the same function and added 1 new one following the same convention as another referrer in the same file.

I'm not sure what the guidance around referrers is but it seems most implementations have their own version, this inconsistency existed before I made any changes.

**Pull request merged**